docs/developer-docs/integrations/sns/managing/making-proposals.md

```diff
@@ -42,6 +42,619 @@ For these cases, SNSs have so called 'generic proposals'. These are custom propo
+                target_canister_id : opt principal;
+                validator_method_name : opt text;
+                target_method_name : opt text;
+            }
+        };
+    };
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            AddGenericNervousSystemFunction = record {
+                id = nat64;
+                name = text;
+                description = opt text;
+                function_type : opt variant {
+                    NativeNervousSystemFunction = record {};
+                    GenericNervousSystemFunction = record {
+                        validator_canister_id = opt principal "ltyfs-qiaab-aaaak-aan3a-cai";
+                        target_canister_id = opt principal;
+                        validator_method_name = opt text;
+                        target_method_name = opt text;
+                    }
+                };
+            }
+        };
+    }
+)'
+```
+
+## `RemoveGenericNervousSystemFunction`
+
+### Relevant type signatures
+
+```candid
+    type RemoveGenericNervousSystemFunction : nat64;
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            RemoveGenericNervousSystemFunction = 123456789:nat64;
+        };
+    }
+)'
+```
+
+## `UpgradeSnsToNextVersion`
+
+### Relevant type signatures
+
+```candid
+    type UpgradeSnsToNextVersion : record {};
+```
+
+### Putting it together
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            UpgradeSnsToNextVersion = record {};
+        };
+    }
+)'
+```
+
+## `RegisterDappCanisters`
+
+### Relevant type signatures
+
+```candid
+    type RegisterDappCanisters : RegisterDappCanisters;
+
+    type RegisterDappCanisters = record { canister_ids : vec principal };
+```
+
+### Putting it together
+
+```candid
+    type RegisterDappCanisters: record {
+        canister_ids : vec principal
+    };
+```
+
+Example in bash:
+
+```bash
+quill sns make-proposal <PROPOSER_NEURON_ID> --proposal '(
+    record {
+        title = "SNS Launch";
+        url = "https://dfinity.org";
+        summary = "A motion to start the SNS";
+        action = opt variant {
+            RegisterDappCanisters = record {
+                canister_ids = vec {principal "ltyfs-qiaaa-aaaak-aan3a-cai", principal "ltyfs-qiaaa-aaaak-aan3a-cai"};
+            };
+        };
+    }
+)'
+```
+
+## `TransferSnsTreasuryFunds`
+
```

Comment:
```suggestion
The SNS DAO has control over a treasury from which funds can be sent to other accounts by `TransferSnsTreasuryFunds` proposals. 
To do so, one has to define the following arguments... 
```

(I am not sure if "from_treasury" defines a ledger canister. I have to clarify this, but think it would be good to list all argument and what they mean here)